src/sentry/utils/compat/pickle.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import, print_function
+
+import kombu.serialization as kombu_serializer
+
+from sentry.utils.compat import pickle
+
+# TODO(python3): We use the pickles `2` protocol as this is the protocol that
+# is supported in both 2 and 3.
+#
+#  - python3 defaults to a protocol > 2 (depending on the version, see [0]).
+#  - python2 defaults to protocol 2.
+#
+# This is ONLY required for the transition of Python 2 -> 3. There will be a
+# period where data may be pickled in python3, where if we did not declare the
+# version, would be in format that python 2's pickle COULD NOT decode.
+#
+# Once the python3 transition is complete we can use a higher version
+#
+# NOTE: from the documentation:
+#       > The protocol version of the pickle is detected automatically
+#
+# [0]: https://docs.python.org/3/library/pickle.html#pickle-protocols
+pickle.DEFAULT_PROTOCOL = 2
+
+
+# TODO(python3): Notes on pickle compatibility between 2 + 3:
+#
+# ## Encoding
+#
+#  In python, when pickling a data structure, the output is a byte stream. We
+#  must tell pickle how to decode byte strings during unpickling that are from
+#  python2 in python3.
+#
+#  Typically we could use the default ASCII decoding, since if correctly used,
+#  byte strings should ONLY contain ASCII data. During the 2 -> 3 conversion,
+#  we should make a great effort to ensure text values are unicode.
```

Comment:
Just did some testing of this for my own understanding.

I make a pickle in py2:
```
In [2]: dumps({'hi': '学中文'})
Out[2]: "(dp0\nS'hi'\np1\nS'\\xe5\\xad\\xa6\\xe4\\xb8\\xad\\xe6\\x96\\x87'\np2\ns."
```

Then try load it in py3
```
In [8]: obj  = pickle.__py3_compat_kombu_pickle_loads(b"(dp0\nS'hi'\np1\nS'\\xe5\\xad\\xa6\\xe4\\xb8\\xad\\xe6\\x96\\x87'\np2\ns.")

In [9]: obj
Out[9]: {'hi': 'å\xad¦ä¸\xadæ\x96\x87'}
```

Then if we use a unicode string
```
In [3]: dumps({'hi': u'学中文'})
Out[3]: "(dp0\nS'hi'\np1\nV\\u5b66\\u4e2d\\u6587\np2\ns."
```

```
In [14]: obj  = pickle.__py3_compat_kombu_pickle_loads(b"(dp0\nS'hi'\np1\nV\\u5b66\\u4e2d\\u6587\np2\ns.")

In [15]: obj
Out[15]: {'hi': '学中文'}
```

This will be tricky if we store pickles like this in the database already, but will be fine for any celery tasks.

Reply:
Yeah for anything in the database that we were incorrectly storing as byte string unicode, we're going to have to come up with some way to either fix, or translate it (That's what I was getting at in a different comment).

It might be hard to spot to be honest :(